pkg/snapshot/v1alpha1/snapshot.go

```diff
@@ -211,6 +211,15 @@ func (s *snapshot) Delete() (*v1alpha1.CASSnapshot, error) {
 		return nil, err
 	}
 
+	// Return success for delete snapshot request in case volume snapshot belongs to a
+	// migrated CSI based persistent volume. Later as part of migration snapshot
+	// can be migrated to CSI snapshots using a migration or manual steps
+	if pv.Spec.CSI != nil {
+		if string(pv.Spec.CSI.Driver) == string(v1alpha1.CSIDriverName) {
+			return nil, nil
```

Comment:
Did we test this scenario? I have a doubt whether we are returning the response code as 200 or not(If object is nil).

Reply:
We are not checking for nil object, only errors should be nil, so this fine.

Reply:
We are checking https://github.com/openebs/maya/blob/c1befaa87a8a708370b9cdc941639d06197a1b9c/cmd/maya-apiserver/app/server/http.go#L450